core/src/main/scala/cats/data/Nested.scala

```diff
@@ -135,13 +135,27 @@ private[data] sealed abstract class NestedInstances9 extends NestedInstances10 {
     }
 }
 
-private[data] sealed abstract class NestedInstances10 {
+private[data] sealed abstract class NestedInstances10 extends NestedInstances11 {
   implicit def catsDataInvariantForNestedContravariant[F[_]: Invariant, G[_]: Contravariant]: Invariant[Nested[F, G, ?]] =
     new NestedInvariant[F, G] {
       val FG: Invariant[λ[α => F[G[α]]]] = Invariant[F].composeContravariant[G]
     }
 }
 
+private[data] sealed abstract class NestedInstances11 extends NestedInstances12 {
+  implicit def catsDataCommutativeApplyForNestedContravariant[F[_]: CommutativeApply, G[_]: CommutativeApply]: CommutativeApply[Nested[F, G, ?]] =
```

Comment:
Usually we place more specific instance higher priority, i.e. `CommutativeApply` and `CommutativeApplicative` should have higher priority than `Apply` and `Applicative`. 
`CommutativeApplicative` should have higher priority than `CommutativeApply`. 

Also line 53 the instance name was wrong `catsDataContravariantForNested` -> `catsDataFunctorForNested` do you mind fix that in this PR as well?